statistics/handle/cache/statscacheinner.go

```diff
@@ -105,10 +121,29 @@ func (sc *StatsCache) PutFromInternal(id int64, t *statistics.Table) {
 	sc.put(id, t, false)
 }
 
+func (sc *StatsCache) putCache(id int64, t *statistics.Table, moveLRUFront bool) bool {
+	ok := sc.c.Put(id, t, moveLRUFront)
+	if ok {
+		return ok
+	}
+	// retry three times and sleep exponentially. 20ms, 40ms and 80ms.
+	for i := 0; i < 3; i++ {
+		time.Sleep(time.Duration(1<<uint(i)) * 10 * time.Millisecond)
+		ok = sc.c.Put(id, t, moveLRUFront)
+		if ok {
+			return ok
+		}
+	}
+	return ok
+}
+
 // Put puts the table statistics to the cache.
 func (sc *StatsCache) put(id int64, t *statistics.Table, moveLRUFront bool) {
-	sc.c.Put(id, t, moveLRUFront)
-
+	ok := sc.putCache(id, t, moveLRUFront)
+	if !ok {
+		logutil.BgLogger().Warn("fail to put the stats cache")
```

Comment:
I suggest we make the log content more detailed here, like adding the table name, stack trace, or anything useful.

Reply:
Done